packages/experimental/Switch/src/Switch.tsx

```diff
@@ -29,19 +31,108 @@ export const switchLookup = (layer: string, state: SwitchState, userProps: Switc
   );
 };
 
+// Function to change background slowly over time when switch is toggled.
+const startTrackBackgroundAnimation = (checked: boolean, animation: Animated.Value) => {
+  const toValue = checked ? 0 : 1;
+  Animated.timing(animation, {
+    toValue,
+    duration: 500,
+    useNativeDriver: false,
+  }).start();
+};
+
+// Function to transform the knob over track in animated way when switch is toggled.
+const startTrackAnimation = (onInit = false, animationConfig: IAnimationConfig, animation: Animated.Value, toggled: boolean) => {
+  Animated.timing(animation, {
+    toValue: toggled
+      ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2)
+      : onInit
+      ? 0
+      : -(animationConfig.trackWidth + animationConfig.thumbWidth),
+    duration: 300,
+    useNativeDriver: false,
+  }).start();
+};
+
+const isMobile = Platform.OS === 'android' || Platform.OS == 'ios';
+
 export const Switch = compose<SwitchType>({
   displayName: switchName,
   ...stylingSettings,
   slots: {
     root: Pressable,
     label: Text,
-    track: View,
-    thumb: View,
+    track: Animated.View, // Conversion from View to Animated.View for Animated API to work
+    thumb: Animated.View,
     toggleContainer: View,
     onOffText: Text,
   },
   useRender: (userProps: SwitchProps, useSlots: UseSlots<SwitchType>) => {
-    const switchInfo = useSwitch(userProps);
+    const switchOnSlot = useSlots(userProps, (layer) => switchLookup(layer, { toggled: true, disabled: userProps.disabled }, {}));
+    const switchOffSlot = useSlots(userProps, (layer) => switchLookup(layer, { toggled: false, disabled: userProps.disabled }, {}));
+
+    const animationConfig: IAnimationConfig = {
+      toggleOnBgColor: switchOnSlot.track({}).props.style.backgroundColor,
```

Comment:
This seems like the Switch will do a lot of work in order to get at the token values. Compressible is probably better for this kind of thing.

Since the value shouldn't change between renders (assuming Switch has no tokensThatAreAlsoProps), you could probably make this once and then just reuse it on every render, minimizing the times that you call React.createElement

Reply:
just a quick question here: I tried with tokensThatAreAlsoProps , seems the token specified also becomes a props and user can pass any value to props that is taken by component as token change and is applied to the token. 

When I try to access it in ,tsx file it doesn't comes as part of userProps or final props. 🤔  , correct me if I am anything wrong.

Reply:
Yeah, your understanding is correct, `tokensThatAreAlsoProps` are component tokens that are exposed as props so that users can change the token value without using `customize`. This is helpful for tokens that are expected to be customized frequently, such as the background color of an Avatar.

The props themselves should be coming through as part of userProps, and are accessed within useStyling when calling useSlots, if I remember correctly.

The way you're doing it now is, I think, the only way to access a token value that is not passed in as a prop. This is where compressible is a better choice as a framework, since you can get direct access to token values inside the render function. You can look at Text as an example.

Reply:
I guess actually you can't assume that the token values will stay constant throughout an instance of the component since interactions can change things like backgroundColor. So I guess this is the way that it has to be if using compose :\

Reply:
Let me try using compressible, seems there're few hacks that is current code of this PR but again it has to do lot of work to just the token value.

Reply:
Also current code having some issue with re-rendering and styling seems the tests starts failing.

Reply:
Taking it to compressible might take some time as there are many things on switch to tackle with animations.

Reply:
Yeah, for now this is fine.

Reply:
Sure , marking this resolved.